### PR TITLE
Switch credentials to pytest-variables

### PR DESCRIPTION
**Pull request by davehunt**
Looks like unvouched and unconfirmed were un*used* so I just removed them and updated the README. Also it looks like the API call is failing but for a different reason than the xfail. I saw the same on our Jenkins instance, so it's unrelated to this patch.

### CODE REVIEW

**Comment by davehunt** [comment]
It turns out this is dependent on updating pytest, which is #191. For now I've pushed an additional commit to this branch, so that we can get some tests running against it.

**Comment by davehunt** [comment]
Passing adhoc job: http://webqa-ci-staging1.qa.scl3.mozilla.com:8080/job/mozillians.adhoc/6/

File: README.md [hunk 1]
```diff
@@ -108,9 +108,10 @@ Some of the tests in mozillians-tests require accounts for
 https://mozillians.allizom.org. You'll need to create three sets of credentials
 with varying privilege levels.
 
-1. Create three username and password combinations on https://mozillians.allizom.org
-2. Join #commtools and ask for one of these users to be vouched (or ask someone on #mozwebqa to do this for you)
-3. Copy mozillians-tests/credentials.yaml to a location outside of mozillians-tests. Update the 'user', 'unvouched', and 'unconfirmed' users in credentials.yaml with those credentials
+1. Create two username and password combinations on https://mozillians.allizom.org
+2. Join #commtools and ask for these users to be vouched (or ask someone on #mozwebqa to do this for you)
+3. In one of these users profile, join at least one group and mark groups as private
+3. Copy mozillians-tests/variables.json to a location outside of mozillians-tests. Update the 'vouched' and 'private' users in variables.json with those credentials
```
Comment (bobsilverberg) [inline README.md]: There are two bullet points labelled as `3.`

**Comment by bobsilverberg** [comment]
Looks good, @davehunt. Just one comment to address. As with PR #195, I get some local failures with this, but the same as I get without it, and none of them seem to be related to these changes.

r+

I'll leave it to you to resolve the conflicts and merge.

**Comment by davehunt** [comment]
Merged in https://github.com/mozilla/mozillians-tests/commit/4de508025772b7a95d3969db292ca9ddea5dc5a1

**Pull request closed without merging**